### DISCUSSION TITLE
Validation dataset in retrain_from_log()

### DISCUSSION THREAD

**Discussion by harshvardhaniimi** [microsoft/FLAML#727] [Q&A]
Hi, I'm trying to retrain a model with LightGBM. Earlier, I trained my model for a period of time and would like the algorithm to pick up where it stopped. `retrain_from_log` allows that, provided I have the log files saved which I do. However, I'm having trouble using the validation dataset with `X_val`. I get the following error.

````
TypeError: fit() got an unexpected keyword argument 'X_val'
````

The code works well when I'm using `automl.fit()`; this error only occurs in `automl.retrain_from_log()`. Any help? Are `X_val` and `Y_val` not valid parameters in `retrain_fromlog()` but only in `fit()`?

**Comment by sonichi** [comment]
`X_val` and `Y_val` are not valid in `retrain_from_log`. Please try removing them.

**Reply by harshvardhaniimi** [reply]
Any reason why it's not implemented? If `fit()` uses custom `X_val` and `y_val`, it's necessary to use it in `retrain_from_log()` to make the logs useful. 

Otherwise, it makes sense to create an issue from this discussion for this feature request. 😄

**Reply by sonichi** [reply]
To clarify, `X_val` and `y_val` can be passed to the underlying estimator if they are supported by the underlying estimator, via `fit_kwargs`. The LightGBM estimator you are using doesn't support them in the `fit()`.

**Reply by harshvardhaniimi** [reply]
I'm using them in my `fit()` function while specifying `"estimator_list": ['lgbm']`, and they worked well without errors...

**Reply by sonichi** [reply]
Yes, it's because `AutoML.fit()` uses `X_val` and `y_val` for validation. But they are not passed to `LGBMEstimator.fit()`.
